### PR TITLE
DX improvements

### PR DESCRIPTION
**Pull request by matt-bernstein**
- switch from pdm to poetry
- improve docker setup, let docker run on local codebase instead of from github
- standardize server API structure for easier communication to LS

### CODE REVIEW

**Comment by matt-bernstein** [comment]
@niklub LMK if you have concerns with any of these changes and I can break them out into separate PRs, if not, can you merge it?

File: server/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# run in Docker
+
+## set environment variables
+
+- OpenAI API key
+- AWS creds:
+```bash
+eval "$(aws configure export-credentials --profile sandbox_dev --format env)"
```
Comment (niklub) [inline server/README.md:8]: `--profile <YOUR-PROFILE>`

File: server/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+cd ui/
+yarn start
+```
+
+# development
+
+## edit client on API change
+
+```python
+import json
```
Comment (niklub) [inline server/README.md:38]: can we pack this in `./api_codegen.sh` ?

File: server/app.py [hunk 3]
```diff
@@ -134,33 +139,34 @@ def get_status(request: JobStatusRequest):
...
-    """
-    job_id: str
-
-
 class JobCancelResponse(BaseModel):
     """
     Response model for cancelling a job.
     """
-    status: str
+    status: Literal['cancelled']
```
Comment (niklub) [inline server/app.py:165]: Why it is not `Status.CANCELED` ?

**Review by niklub — CHANGES_REQUESTED** [review]

**Comment by niklub** [comment]
Overall, LGTM with a few minor changes

**Comment by matt-bernstein** [comment]
thanks for the review, made the changes you suggested and will merge when this github outage is over and the PR updates :)

**Pull request merged**